kinder/pkg/cluster/manager/actions/setup-external-ca.go

```diff
@@ -0,0 +1,114 @@
+)
+
+// SetupExternalCA setups certificates and kubeconfig files to be able to create a cluster without CA keys.
+func SetupExternalCA(c *status.Cluster, vLevel int) error {
+	fmt.Println("Setuping external CA for the cluster...")
+
+	// gets the IP of the load balancer
+	loadBalancerIP, _, err := c.ExternalLoadBalancer().IP()
+	if err != nil {
+		return errors.Wrapf(err, "failed to get IP for node: %s", c.ExternalLoadBalancer().Name())
+	}
+
+	// generate certs on the primary node
+	if err := c.BootstrapControlPlane().Command(
+		"/bin/sh", "-c",
+		fmt.Sprintf("kubeadm init phase certs all --control-plane-endpoint=%s --v=%d",
+			loadBalancerIP,
+			vLevel),
+	).RunWithEcho(); err != nil {
+		return errors.Wrapf(err, "could not generate certs on node: %s", c.BootstrapControlPlane().Name())
+	}
+
+	// copy the shared certs to the rest of the CP nodes
+	for _, n := range c.SecondaryControlPlanes() {
+		if err := copyCertificatesToNode(c, n); err != nil {
+			return errors.Wrapf(err, "could not copy certs to node %s", n.Name())
+		}
+	}
+
+	// generate kubeconfig files on the primary node
+	if err := c.BootstrapControlPlane().Command(
+		"/bin/sh", "-c",
+		fmt.Sprintf("kubeadm init phase kubeconfig all --control-plane-endpoint=%s --v=%d",
+			loadBalancerIP,
+			vLevel),
+	).RunWithEcho(); err != nil {
+		return errors.Wrapf(err, "could not generate kubeconfig files on node: %s", c.BootstrapControlPlane().Name())
+	}
+
+	// copy the shared kubeconfig files to the rest of the CP nodes
+	for _, n := range c.SecondaryControlPlanes() {
+		if err := copyKubeconfigFilesToNode(c, n); err != nil {
+			return errors.Wrapf(err, "could not copy kubeconfig files to node %s", n.Name())
+		}
+	}
+
+	// generate remaining certificates on the secondary CP nodes
+	for _, n := range c.SecondaryControlPlanes() {
+		if err := n.Command(
+			"/bin/sh", "-c",
+			fmt.Sprintf("kubeadm init phase certs all --control-plane-endpoint=%s --v=%d",
+				loadBalancerIP,
+				vLevel),
+		).RunWithEcho(); err != nil {
+			return errors.Wrapf(err, "could not generate remaining certificates on node: %s", n.Name())
+		}
+	}
+
+	// iterate all nodes other than the primary CP node
+	for _, n := range append(c.SecondaryControlPlanes(), c.Workers()...) {
+		// copy the CA cert/key to workers. CP nodes should already have them.
+		if !n.IsControlPlane() {
+			if err := copyCAToNode(c, n); err != nil {
+				return err
+			}
+		}
+
+		// generate a kubelet.conf using the CA
+		if err := n.Command(
+			"/bin/sh", "-c",
+			fmt.Sprintf("kubeadm init phase kubeconfig kubelet --control-plane-endpoint=%s --v=%d",
+				loadBalancerIP,
+				vLevel),
+		).RunWithEcho(); err != nil {
+			return errors.Wrapf(err, "could not generate a kubelet.conf on node: %s", n.Name())
+		}
+
+		// delete the ca.key
+		if err := n.Command("rm", "-f", "/etc/kubernetes/pki/ca.key").Run(); err != nil {
+			return errors.Wrapf(err, "could not delete ca.key on node: %s", n.Name())
+		}
```

Comment:
What about having a final loop deleting ca.key from all nodes instead of having the operation being repeated in two places